clients/js/test/plugins/asset/permanentFreezeExecute.test.ts

```diff
@@ -0,0 +1,460 @@
+import { transferSol } from '@metaplex-foundation/mpl-toolbox';
+import {
+  createNoopSigner,
+  generateSigner,
+  publicKey,
+  sol,
+} from '@metaplex-foundation/umi';
+import test from 'ava';
+
+import {
+  addPluginV1,
+  burnV1,
+  create,
+  createPlugin,
+  execute,
+  fetchAssetV1,
+  findAssetSignerPda,
+  PluginType,
+  removePluginV1,
+  transferV1,
+  updatePluginV1,
+} from '../../../src';
+import {
+  assertAsset,
+  assertBurned,
+  createAsset,
+  createUmi,
+  DEFAULT_ASSET,
+} from '../../_setupRaw';
+
+test('it can freeze and unfreeze execute with PermanentFreezeExecute', async (t) => {
+  // Given a Umi instance
+  const umi = await createUmi();
+  const owner = generateSigner(umi);
+  await umi.rpc.airdrop(owner.publicKey, sol(1));
+
+  // Create an asset with PermanentFreezeExecute plugin (frozen by default)
+  const assetSigner = generateSigner(umi);
+
+  await create(umi, {
+    asset: assetSigner,
+    owner: owner.publicKey,
+    name: 'Test Asset',
+    uri: 'https://example.com/asset',
+    plugins: [
+      {
+        type: 'PermanentFreezeExecute',
+        frozen: true,
+      },
+    ],
+  }).sendAndConfirm(umi);
+
+  const asset = await fetchAssetV1(umi, publicKey(assetSigner));
+
+  await assertAsset(t, umi, {
+    ...DEFAULT_ASSET,
+    asset: asset.publicKey,
+    owner: owner.publicKey,
+    updateAuthority: { type: 'Address', address: umi.identity.publicKey },
+    permanentFreezeExecute: {
+      authority: {
+        type: 'UpdateAuthority',
+      },
+      frozen: true,
+    },
+  });
+
+  // Fund the asset signer PDA
+  const [assetSignerPda] = findAssetSignerPda(umi, {
+    asset: asset.publicKey,
+  });
+
+  await transferSol(umi, {
+    source: umi.identity,
+    destination: publicKey(assetSignerPda),
+    amount: sol(0.5),
+  }).sendAndConfirm(umi);
+
+  // Attempt Execute → should fail because plugin is frozen
+  const recipient = generateSigner(umi);
+
+  const execResult = execute(umi, {
+    asset,
+    payer: owner,
+    instructions: transferSol(umi, {
+      source: createNoopSigner(publicKey(assetSignerPda)),
+      destination: recipient.publicKey,
+      amount: sol(0.1),
+    }),
+  }).sendAndConfirm(umi);
+
+  await t.throwsAsync(execResult, { name: 'InvalidAuthority' });
+
+  // Unfreeze the execute with update authority
+  await updatePluginV1(umi, {
+    asset: asset.publicKey,
+    plugin: createPlugin({
+      type: 'PermanentFreezeExecute',
+      data: { frozen: false },
+    }),
+  }).sendAndConfirm(umi);
```

Comment:
Nit: I think you have the same authority for botht he asset owner and the update authority, they are both umi identity.  Might be good to have a test that only udpate authority can do these things and not owner